src/test/java/spoon/test/architecture/SpoonArchitectureEnforcerTest.java

```diff
@@ -206,4 +208,91 @@ public boolean matches(CtClass element) {
 			assertTrue(klass.getElements(new TypeFilter<>(CtConstructor.class)).stream().allMatch(x -> x.hasModifier(ModifierKind.PRIVATE)));
 		}
 	}
+
+	@Test
+	public void testSpecPackage() throws Exception {
+		// contract: when a pull-request introduces a new package, it is made explicit during code review
+		// when a pull-request introduces a new package, this test fails and the author has to explicitly declare the new package here
```

Comment:
It could be great to improve CONTRIBUTING.md with this kind of contract, WDYT?

Reply:
just do not make  CONTRIBUTING.md too complex, so the new contributors does not have fear to contribute ;-)

Reply:
excellent point Pavel! I'll make a PR in that direction.